src/components/Calculator.js

```diff
@@ -1,34 +1,152 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../App.css';
 import Button from './Button';
+import calculate from '../logic/Calculate';
 
-const Calculator = () => (
-  <div className="main">
-    <div className="calculator">
-      <div className="screen">0</div>
-      <div className="input-container">
-        <Button type="button" text="AC" className="inputs" />
-        <Button type="button" text="+/-" className="inputs" />
-        <Button type="button" text="%" className="inputs" />
-        <Button type="button" text="/" className="inputs-right" />
-        <Button type="button" text="7" className="inputs" />
-        <Button type="button" text="8" className="inputs" />
-        <Button type="button" text="9" className="inputs" />
-        <Button type="button" text="x" className="inputs-right" />
-        <Button type="button" text="4" className="inputs" />
-        <Button type="button" text="5" className="inputs" />
-        <Button type="button" text="6" className="inputs" />
-        <Button type="button" text="-" className="inputs-right" />
-        <Button type="button" text="1" className="inputs" />
-        <Button type="button" text="2" className="inputs" />
-        <Button type="button" text="3" className="inputs" />
-        <Button type="button" text="+" className="inputs-right" />
-        <Button type="button" text="0" className="zero-input" />
-        <Button type="button" text="." className="dot-inputs" />
-        <Button type="button" text="=" className="equal-inputs" />
+const Calculator = () => {
+  const [display, setDisplay] = useState('0');
+
+  const [calculatorData, setCalculatorData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleButtonClick = (buttonName) => {
+    const newData = calculate(calculatorData, buttonName);
+    setCalculatorData(newData);
+
+    if (newData.next !== null) {
+      setDisplay(newData.next);
+    } else if (newData.total !== null) {
+      setDisplay(newData.total);
+    } else {
+      setDisplay('0');
+    }
+  };
+  return (
+    <div className="main">
+      <div className="calculator">
+        <div className="screen">{display}</div>
+        <div className="input-container">
+          <Button
+            type="button"
+            text="AC"
+            className="inputs"
+            onClick={() => handleButtonClick('AC')}
+          />
+          <Button
+            type="button"
+            text="+/-"
+            className="inputs"
+            onClick={() => handleButtonClick('+/-')}
+          />
+          <Button
+            type="button"
+            text="%"
+            className="inputs"
+            onClick={() => handleButtonClick('%')}
+          />
+          <Button
+            type="button"
+            text="/"
+            className="inputs-right"
+            onClick={() => handleButtonClick('/')}
+          />
```

Comment:
Great job so far @Developer-Kingz but you still have some little fix here.

- [ ] Please your app throws an error when the user tries to use the divisible operator. 
![error](https://github.com/Developer-Kingz/maths-magician/assets/55596970/6b2b8500-9828-4a9b-8c6f-bbac6d19fc72)